ui/desktop/app/routes/scopes/scope/projects/targets/index.js

```diff
@@ -183,6 +185,27 @@ export default class ScopesScopeProjectsTargetsIndexRoute extends Route {
         // the page in case the controller doesn't support aliases yet
       }
 
+      // Remove expired sessions in ember data store for symmetry with cache.
+      if (allAssociatedSessions) {
+        const targetIds = targets.map(({ id }) => id);
+        const allAssociatedSessionIds = new Set(
+          allAssociatedSessions.map(({ id }) => id),
+        );
+        const storedSessionIds = new Set(
+          this.store
+            .peekAll('session')
+            .filter((s) => targetIds.includes(s?.target_id))
+            .map(({ id }) => id),
+        );
+        const removedSessions = storedSessionIds.difference(
+          allAssociatedSessionIds,
+        );
+        removedSessions.forEach((id) => {
+          const record = this.store.peekRecord('session', id);
+          this.store.unloadRecord(record);
+        });
+      }
+
```

Comment:
I removed the `unloadAll('session')` and opted to removing only particular records for the loaded page (max 50 targets per page) instead since unloading all sessions would also unload attached proxy details for each session even if its still active/pending.